src/interpolations.jl

```diff
@@ -0,0 +1,49 @@
+# FIXME: upstream https://github.com/JuliaGraphics/ColorVectorSpace.jl/issues/75
+@inline _nan(::Type{HSV{Float16}}) = HSV{Float16}(NaN16,NaN16,NaN16)
+@inline _nan(::Type{HSV{Float32}}) = HSV{Float32}(NaN32,NaN32,NaN32)
+@inline _nan(::Type{HSV{Float64}}) = HSV{Float64}(NaN,NaN,NaN)
+@inline _nan{T}(::Type{T}) = nan(T)
+
+# The default values used by extrapolation for off-domain points
+@compat const FillType = Union{Number,Colorant,Flat,Periodic,Reflect}
+@compat const FloatLike{T<:AbstractFloat} = Union{T,AbstractGray{T}}
+@compat const FloatColorant{T<:AbstractFloat} = Colorant{T}
+@inline _default_fill{T<:FloatLike}(::Type{T}) = convert(T, NaN)
+@inline _default_fill{T<:FloatColorant}(::Type{T}) = _nan(T)
+@inline _default_fill{T}(::Type{T}) = zero(T)
+
+box_extrapolation(etp::AbstractExtrapolation) = etp
```

Comment:
Just that the image is "boxed" or "decorated" by an extrapolation. It's supposed to signify that this will be a view.

Can you think of a better name?